File: SP-UI/src/app/models/department.ts
import { Category, MenuCategory, PageCategory } from "./category"

export interface Department {
    id: number
    name: string
    description: string
}

export interface MenuDepartment {
    id: number
    name: string,
    categories: MenuCategory[]
}

export interface PageDepartment {
    id: number
    name: string
    description: string
    categories: PageCategory[]
}

File: SP-API/SP-API/Entities/Department.cs
using System.ComponentModel.DataAnnotations;

namespace Mall_API.Entities
{
    public class Department
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Description { get; set; }
        public ICollection<Category> Categories { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}
